### PR TITLE
Split the measurement tool into lines and polygons

### PR DESCRIPTION
**Pull request by glughi**
### What this PR does

Split the measurement tool into lines and polygons.
You can now add a point to an existing segment.
Lines and polygons are clamped to the ground (this can be changed using `clampMeasureLineToGround` in `Terria.ts`).

image

### Checklist

- [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] I've updated relevant documentation in `doc/`.
- [x] I've updated CHANGES.md with what I changed.
- [ ] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/Models/UserDrawing.ts [hunk 1]
```diff
@@ -346,6 +381,39 @@ export default class UserDrawing extends MappableMixin(
     }
   }
 
+  private insertPointToPointEntities(
+    name: string,
+    position: Cartesian3,
+    index: number
+  ) {
+    var pointEntity = new Entity({
```
Comment (pjonsson) [inline lib/Models/UserDrawing.ts]: Can this be const?

File: lib/Models/UserDrawing.ts [hunk 2]
```diff
@@ -426,6 +494,41 @@ export default class UserDrawing extends MappableMixin(
           }
           if (isDefined(pickedFeatures.pickPosition)) {
             const pickedPoint = pickedFeatures.pickPosition;
+            const picketCarto = Cartographic.fromCartesian(pickedPoint);
```
Comment (pjonsson) [inline lib/Models/UserDrawing.ts]: Is `picketCarto` meant to be `pickedCarto`?

File: lib/Models/UserDrawing.ts [hunk 3]
```diff
@@ -483,9 +596,17 @@ export default class UserDrawing extends MappableMixin(
...
+              ? new Color(0.0, 0.666, 0.843, 0.25)
+              : new Color(0.0, 0.666, 0.843, 0),
             outlineColor: new Color(1.0, 1.0, 1.0, 1.0),
-            perPositionHeight: true as any
+            //perPositionHeight: true as any
+            // Clamp to ground polygons of Measure Tool
+            heightReference: new ConstantProperty(
+              HeightReference.CLAMP_TO_GROUND
+            ),
+            perPositionHeight: <any>false
```
Comment (pjonsson) [inline lib/Models/UserDrawing.ts]: Are the commented out lines of code leftovers from debugging?

The previous code for `perPositionHeight` is `true as any`, does `false as any` not work?

File: lib/ReactViews/Map/MapNavigation/Items/MeasureLineTool.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    this.updateDistance(pointEntities);
+  }
+
+  onPointMoved(pointEntities: CustomDataSource) {
+    // This is no different to clicking a point.
+    this.onPointClicked(pointEntities);
+  }
+
+  onMakeDialogMessage = () => {
+    const distance = this.prettifyNumber(this.totalDistanceMetres, false);
```
Comment (pjonsson) [inline lib/ReactViews/Map/MapNavigation/Items/MeasureLineTool.ts]: Can't this be `return this.prettifyNumber(...)` directly without the extra intermediate variables?

File: lib/Models/UserDrawing.ts (hunk 1 above)
Comment (glughi) [reply lib/Models/UserDrawing.ts]: Yes.

**Pull request closed without merging**